### DISCUSSION TITLE
Why to define MSRV, why not use Stable Rust.

### DISCUSSION THREAD

**Discussion by sanjeevprasad** [helix-editor/helix#9142] [Q&A] [answered]
Hi helix has Minimum Stable Rust Version (MSRV) Policy which actually follows Firefox's Policy.
I just want to know why can't we just use latest stable version of Rust, as we are not exposing anything publicly (I think so may be I am wrong.).

I am new to the repo, I just wanted to contribute to helix, by fixing some of the issue. Sorry for opening a new one without fixing any, because this is the first thing I noticed.

Because by limiting ourselves to previous versions of Rust prevent us from using new features like https://blog.rust-lang.org/2023/12/21/async-fn-rpit-in-traits.html

I know this may not be the main focus of the project, but It will be nice, because later on we have to refactor less.
and we may be able to use ```impl SomeTrait```, instead of ```Box<dyn Trait>```.

**Comment by the-mikedavis** [comment, marked as answer]
It makes it possible for packagers to build Helix without being on the latest possible Rust. For example Ubuntu is usually a few versions behind the latest and tends to align with Firefox's MSRV.

**Reply by sanjeevprasad** [reply]
I understand, but Most of the time we use rustup to install rust.
For a language like c or c++ it is makes sense to use official ones, because it is cumbersome to install it manually. 
But for a fast paced language like Rust. I think we should use latest stable release.
also ```Box<Something>``` is slower, and more than that it is really Ugly.

**Reply by the-mikedavis** [reply]
Individuals might use rustup but package managers tend not to.

We'll take advantage of the new features as we bump the MSRV according to the policy - there isn't a rush.

**Reply by sanjeevprasad** [reply]
Just curious, is rustup really necessary, doesn't cargo automatically installs latest rust version, even if installed from package manager, if defined latest in rust-toolchain.toml
```toml
[toolchain]
channel = "stable"
components = ["rustfmt", "rust-src", "clippy"]
```

**Reply by pascalkuthe** [reply]
that is rustup functionality, cargo does not do that automatically

**Reply by the-mikedavis** [reply]
yeah, the toolchain file is a part of rustup rather than cargo https://rust-lang.github.io/rustup/overrides.html#the-toolchain-file

rustup basically makes `cargo` a shim that checks the version and then delegates to the actual `cargo` binary for the specified version

**Reply by sanjeevprasad** [reply]
Ok cool. let's stick with whatever we have right now!
